src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -4315,6 +4315,8 @@ void MacroAssembler::load_method_holder(Register holder, Register method) {
 }
 
 void MacroAssembler::load_klass(Register dst, Register src) {
+  assert(oopDesc::klass_offset_in_bytes() < static_cast<intptr_t>(os::vm_page_size()),
+         "Doesn't need explicit null check");
```

Comment:
Sorry I meant for this to be a "global" assert somewhere, executed once per VM startup, not done on each call like this.

Reply:
Suggest a global place please and but not in oopDesc::klass_offset_in_bytes() because that would require oop.hpp to include os.hpp.  I couldn't find somewhere good where we wouldn't have to write a large comment to explain the context.

Reply:
Perhaps in `Universe::genesis()` in file universe.cpp? That is where all sorts of hand-cranked Klass-related gubbins gets set up so it seems like a sensible place to check before the create. Likewise for the array offset assert.

WDYT, Coleen/David?

Reply:
That sounds okay to me.

Reply:
Sure.